### ISSUE TITLE
llvm-13 Error: bad register name `%eb'

### ISSUE THREAD

**Issue by juanitotc** [llvm/llvm-project#53283]
Building llvm-13.0.0 on a 32-bit linux os using gcc-11.2 and glibc-2.34 using:

```
cmake -LAH -DCMAKE_C_FLAGS_RELEASE="-march=pentium-m -mtune=i686 -Os -pipe" -DCMAKE_CXX_FLAGS_RELEASE="-march=pentium-m -mtune=i686 -Os -pipe" -DCMAKE_INSTALL_PREFIX=/usr/local -DLLVM_ENABLE_FFI=ON -DLLVM_BUILD_LLVM_DYLIB=ON -DLLVM_LINK_LLVM_DYLIB=ON -DLLVM_TARGETS_TO_BUILD="host;AMDGPU;BPF" -DLLVM_ENABLE_RTTI=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE_MAKEFILE=TRUE -DLLVM_BINUTILS_INCDIR=/usr/local/include -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_LIBEXECDIR=lib/clang -Wno-dev -G Ninja ../
```

..fails with:

```
FAILED: lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/SelectionDAGBuilder.cpp.o 
/usr/local/bin/c++  -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/CodeGen/SelectionDAG -I../lib/CodeGen/SelectionDAG -Iinclude -I../include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wmisleading-indentation -fdiagnostics-color -ffunction-sections -fdata-sections -march=pentium-m -mtune=i686 -Os -pipe    -fno-exceptions -std=c++14 -MD -MT lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/SelectionDAGBuilder.cpp.o -MF lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/SelectionDAGBuilder.cpp.o.d -o lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/SelectionDAGBuilder.cpp.o -c ../lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp
{standard input}: Assembler messages:
{standard input}:54880: Warning: end of file not at end of a line; newline inserted
{standard input}:55525: Error: bad register name `%eb'
{standard input}: Error: open CFI at the end of file; missing .cfi_endproc directive
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
```

This also fails with "-march=i686"

Note that build succeeds on a 64-bit linux os using "-march=generic" with gcc-11.2 and glibc-2.34 and succeeded with llvm-8.0.0 on an earlier version of the same 32-bit linux os

**Comment by topperc** [comment]
This sounds more like a gcc or binutils assembler issue than an llvm issue. gcc compiles source files to assembly and gives them to the binutils assembler to produce machine code. That error reads like the assembly output from gcc is incomplete.

You didn't run out of disk space did you?

**Comment by juanitotc** [comment]
This os (tinycorelinux) runs in ram so, in case 4gb wasn't enough, I tried building on permanent storage, but the build still failed with the same error.

In ram, the build fails at ninja command +/- 500, whereas in permanent storage it fails at ninja command +/- 3000.

Without changing anything, I ran the ninja command again and the remainder of the build completed without error.

I'm guessing that ninja doesn't tidy up after itself whilst running and gradually fills ram with temporary files - perhaps using -j1 would help, but at no point did I get an out of memory error.
